src/exoscale/compute/api/http.clj

```diff
@@ -25,21 +29,21 @@
 
 (def default-http-opts
   "Default HTTP options passed to the underlying HTTP library (aleph)."
-  {:connection-timeout 10000
-   :request-timeout    10000
-   :pool-timeout       1000
-   :read-timeout       10000})
```

Comment:
For compatibility, we need to support those timeout params. 
For example they are used in [public-api](https://github.com/exoscale/public-api/blob/master/resources/config.edn#L146-L149). They are forwarded to [alpeh http-client](https://github.com/aleph-io/aleph/blob/master/src/aleph/http.clj#L282-L285).

`connection-timeout` can be replaced by [`java.net.http.HttpClient.Builder/connectTimeout`](https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpClient.Builder.html#connectTimeout(java.time.Duration))
`request-timeout` can be replaced by [`java.net.http.http.HttpRequest.Builder/timeout`](https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpRequest.Builder.html#timeout(java.time.Duration))

I am pretty sure we do not rely on `pool-timeout`. I cannot think of scenarios where this timeout can be of use.

I am more concerned by `read-timeout` which allow us to handle scenario where server is having hard time. 
Maybe we should express it via `auspex/timeout` on top of the interceptor pipeline

Reply:
There's a compatibility layer to support these alrrady, I actually need to restore these defaults now. (will do that tomorrow, I am off this afternoon)

Reply:
As mentioned it's already here: 

https://github.com/exoscale/clojure-exoscale/pull/19/files/db2fd9813e9a46ae1a082a5a50946280c1869b19#diff-3ef73edee83f24dd9cf427aa6a7789e195b8bd463a366fba044d8dcbb0f6cd62R12

https://github.com/exoscale/clojure-exoscale/pull/19/files/db2fd9813e9a46ae1a082a5a50946280c1869b19#diff-3ef73edee83f24dd9cf427aa6a7789e195b8bd463a366fba044d8dcbb0f6cd62R32

The one difference with the existing code is that connect-timeout is bound to the client instance not the request, but that's an acceptable trade-off (you can always spin your own version of the client and use it per call).

Reply:
`connection-timeout` used to be define in `http-opt` and was per request.
With this change it becomes a client configuration => we will need to create custom client and use it when calling `request!!` `config` param
I think your approach is valid (better to create client once than in every request) but we need to communicate this change so that we ensure we migrate dependent projects

Reply:
Connection timeouts should be quite uncommon and there is little reason to make it vary from calls to different endpoints imho. Request timeout is way more important per call. 
But yes we can just spin a client instance on our component and use it directly (we'll need to do thay for ssl internally anyway).